lib/Sema/CodeSynthesis.cpp

```diff
@@ -351,34 +351,59 @@ static ConstructorDecl *createImplicitConstructor(NominalTypeDecl *decl,
     // and actor initializers do not run on the actor, so initial values
     // cannot be actor-instance-isolated.
     bool shouldAddNonisolated = true;
-    llvm::Optional<ActorIsolation> existingIsolation = llvm::None;
+    ActorIsolation existingIsolation = getActorIsolation(decl);
     VarDecl *previousVar = nullptr;
 
-    // The memberwise init properties are also effectively what the
-    // default init uses, e.g. default initializers initialize via
-    // properties wrapped and init accessors.
-    for (auto var : decl->getMemberwiseInitProperties()) {
+    for (auto member : decl->getImplementationContext()->getAllMembers()) {
+      auto pbd = dyn_cast<PatternBindingDecl>(member);
+      if (!pbd || pbd->isStatic())
+        continue;
+
+      auto *var = pbd->getSingleVar();
+      if (!var) {
+        shouldAddNonisolated = false;
```

Comment:
I don't understand why a pattern without a single variable should suppress `nonisolated`.

Reply:
Oh, it shouldn't. I skipped them because I thought they were banned, but we only ban stored properties without a single var for struct `let`s. I'll fix this, thank you!

Reply:
Fixed in https://github.com/apple/swift/pull/71033/commits/cc1fd1ce8fb13518f8d7f42e37c2e995d03c3b23